node-graph/gcore/src/application_io.rs

```diff
@@ -112,10 +112,25 @@ impl<T: ApplicationIo> ApplicationIo for &T {
 	}
 }
 
+#[derive(Debug, Clone)]
+pub enum NodeGraphUpdateMessage {
+	ImaginateStatusUpdate,
+}
+
+pub trait NodeGraphUpdateSender {
+	fn send(&self, message: NodeGraphUpdateMessage);
+}
+
+pub trait GetImaginatePreferences {
+	fn get_host_name(&self) -> &str;
+}
+
```

Comment:
would it make more sense to put this into the application io trait?

Reply:
I don't know. There are no docs explaining what ApplicationIo and EditorApi actually mean. Would you explain it to me?

Reply:
the idea is to provide a platform independent interface for interacting with the outside world + Provide information about the editor state. we currently store the editor state stuff in EditorApi and facilitate the io interaction through the ApplicationIo trait but we might restructure this anyway in the future so it probably does not really matter for now